### ISSUE TITLE
Unable to use fill-column-indicator

### ISSUE THREAD

**Issue by mandarvaze** [hlissner/doom-emacs#662]
### Observed behavior

`package-list-packages` shows the fill-column-indicator is installed

```
fill-column-indicator is an installed package.
     Status: Installed in ‘fill-column-indicator-20171209.1124/’ 
```

But I do not get `fci-` related functions when I type `M-x`
`turn-on-fci-mode` in my `config.el` also does not take effect (Duh!)

### Expected behavior

I should be able to show vertical line at column 80

I do **not** want `visual-fill-column-mode`

### Steps to reproduce

No specific steps.

But I read that `fill-column-indicator` is incompatible with `company-mode` (Which I am using)
But I am not sure that applies here, cause it talks about pop-up appearing *behind* the column-indicator or something. In my case, I can't even see the `fci-` options.

I've also read that `fill-column-indicator` is incompatible with `whitespace-mode`, I do not understand whether it applies in my case.

Finally, I was able to use `fci-`, `company` and `whitespace-mode` together in spacemacs.

### System information

<details>
<summary>Click to expand</summary>

```
- OS: darwin (x86_64-apple-darwin16.7.0)
- Emacs: 26.1 (May 29, 2018)
- Doom: 2.0.9 (develop 530bdca0977f2cd145e497054f00a72b7c80e306)
- Graphic display: t (daemon: nil)
- System features: JPEG RSVG IMAGEMAGICK NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS MODULES THREADS LCMS2
- Details:
  ```elisp
  uname -a:  Darwin MacBook-Pro.local 16.7.0 Darwin Kernel Version 16.7.0: Tue Jan 30 11:27:06 PST 2018; root:xnu-3789.73.11~1/RELEASE_X86_64 x86_64
  modules:   (:feature (evil +everywhere) file-templates (lookup +devdocs +docsets) snippets spellcheck (syntax-checker +childframe) version-control workspaces :completion (helm +fuzzy) :ui doom-dashboard doom-modeline doom-quit evil-goggles hl-todo nav-flash neotree (popup +all +defaults) vi-tilde-fringe window-select :emacs ediff electric-indent eshell imenu :tools magit pdf :lang elixir emacs-lisp javascript markdown (org +attach +babel +capture +export +present) python sh solidity web :app (write +wordnut +langtool) :config)
  packages:  (ace-link ace-window alchemist all-the-icons anaconda-mode anzu async auto-compile auto-yasnippet avy (centered-window :recipe (centered-window :fetcher github :repo anler/centered-window-mode)) coffee-mode command-log-mode company company-anaconda company-dict company-shell company-web csv-mode (devdocs-lookup :recipe (devdocs-lookup :fetcher github :repo skeeto/devdocs-lookup)) dired-k dockerfile-mode doom-themes dtrt-indent dumb-jump elixir-mode (emacs-snippets :recipe (emacs-snippets :fetcher github :repo hlissner/emacs-snippets :files (*))) emmet-mode eslintd-fix esup evil evil-anzu evil-args evil-collection evil-commentary evil-easymotion evil-ediff evil-embrace evil-escape evil-exchange evil-goggles evil-indent-plus evil-magit evil-matchit evil-mc evil-multiedit evil-numbers evil-org evil-snipe evil-surround evil-textobj-anyblock evil-vimish-fold evil-visualstar (exec-path-from-shell :ignore nil) expand-region flycheck flycheck-cask flycheck-popup-tip flycheck-posframe flyspell-correct flyspell-correct-helm git-gutter-fringe git-link git-timemachine gitconfig-mode gitignore-mode graphql-mode haml-mode helm helm-ag helm-c-yasnippet helm-company helm-css-scss helm-dash (helm-describe-modes :recipe (helm-describe-modes :fetcher github :repo emacs-helm/helm-describe-modes)) helm-flx helm-projectile helm-swoop helm-xref helpful hide-mode-line highlight-indentation highlight-numbers highlight-quoted hl-todo htmlize hydra imenu-anywhere imenu-list js2-mode js2-refactor json-mode langtool less-css-mode macrostep magit magit-gitflow magithub markdown-mode markdown-toc mixed-pitch nav-flash neotree nodejs-repl nose ob-mongo ob-sql-mode ob-translate (org-bullets :recipe (org-bullets :fetcher github :repo Kaligule/org-bullets)) org-download org-plus-contrib org-tree-slide (osx-clipboard :ignore nil) overseer ox-pandoc ox-reveal pcre2el pdf-tools persistent-soft persp-mode pip-requirements projectile pug-mode quelpa quickrun rainbow-delimiters rainbow-mode restart-emacs rjsx-mode sass-mode shrink-path skewer-mode slim-mode slime smartparens solaire-mode solidity-mode stylus-mode synosaurus tide toc-org toml-mode twittering-mode typescript-mode undo-tree use-package vi-tilde-fringe vimrc-mode visual-fill-column web-beautify web-mode wgrep which-key wordnut xref-js2 yaml-mode yasnippet)
  elc dirs:  n/a
  exec-path: (/usr/local/bin/ /usr/bin/ /bin/ /usr/sbin/ /sbin/ /Applications/Wireshark.app/Contents/MacOS/ /projects/gopath/bin/ ~/.local/bin/ /usr/local/Cellar/emacs-plus/26.1/libexec/emacs/26.1/x86_64-apple-darwin16.7.0/)
```

**Comment by hlissner** [comment]
At minimum, you must have this in your `~/.doom.d/packages.el`:

```elisp
(package! fill-column-indicator)
```

Then run `make` (or `bin/doom refresh`, they are equivalent) to make sure everything is installed and your autoloads files are up to date

After that, `fci-mode` and `turn-on-fci-mode` will be visible to you from the `M-x` menu. If you'd like to always keep it on, I'd suggest adding this to your `~/.doom.d/config.el`:

```elisp
(def-package! fci-mode
  :after-call doom-before-switch-buffer-hook
  :config
  (defvar-local company-fci-mode-on-p nil)

  (defun company-turn-off-fci (&rest ignore)
    (setq company-fci-mode-on-p fci-mode)
    (when fci-mode (fci-mode -1)))

  (defun company-maybe-turn-on-fci (&rest ignore)
    (when company-fci-mode-on-p (fci-mode 1)))

  (add-hook 'company-completion-started-hook #'company-turn-off-fci)
  (add-hook 'company-completion-finished-hook #'company-maybe-turn-on-fci)
  (add-hook 'company-completion-cancelled-hook #'company-maybe-turn-on-fci))
```

This will disable fci-mode only while company is activate.

Let me know if that works for you!

**Comment by mandarvaze** [comment]
The first part worked. i.e. I now have `fci-mode` available via `M-x`
I have put `packages.el` in my DOOMDIR (I'm on develop) i.e. `~/doom-config`

I had originally installed the package via "normal" method, which doesn't work for doom (Duh!)

As for the second part - I copied the configuration "as-is" to the `$DOOMDIR/config.el` - But I get error "No such file or directory : fci-mode" randomly (e.g. when I did `SPC b B`)
Commenting that seems to make the error go away.

**Comment by hlissner** [comment]
Oh, my bad! There was a mistake in my snippet: the package is called `fill-column-indicator`, not `fci-mode`!

In any case, I've just pushed a `:ui fci` module which includes this functionality. Upgrade to the latest Doom and enable it in your `$DOOMDIR/init.el`.

**Comment by mandarvaze** [comment]
Awesome - works well.
One more suggestion is that we may wish to turn if OFF for the org-mode.
Due to the how the org-mode files are rendered (bigger fonts for headers, and indented text for lower level, it **appears** that the text has flown past column 80 - but it has **not**. The column-indicator also *looks* broken line rather than one continuous straight line.

<img width="717" alt="org-mode-fci" src="https://user-images.githubusercontent.com/46438/41157342-1307e1ca-6b43-11e8-85b2-9ebfe406eb93.png">

I used [this](http://members.optusnet.com.au/~charles57/GTD/orgmode.org) file 
Then I used `fill-paragraph` to automatically "break" the lines at 79th column , to show case the problem. (Original file had paragraphs broken at 69th or so column - so "out of the box" there was no problem.)

Using `fill-paragraph` to automatically "break" the lines at 79th column works for  text-mode and prog-mode, but cause the problem shown above for the org-mode.
